CRM/Report/Form.php

```diff
@@ -2637,23 +2643,22 @@ public function buildQuery($applyLimit = TRUE) {
    * Build group by clause.
    */
   public function groupBy() {
-    $groupBys = array();
     if (!empty($this->_params['group_bys']) &&
       is_array($this->_params['group_bys'])
     ) {
       foreach ($this->_columns as $tableName => $table) {
         if (array_key_exists('group_bys', $table)) {
           foreach ($table['group_bys'] as $fieldName => $field) {
             if (!empty($this->_params['group_bys'][$fieldName])) {
-              $groupBys[] = $field['dbAlias'];
+              $this->_groupByArray[] = $field['dbAlias'];
```

Comment:
(the vast majority of reports that have groupBy functions override them for no reason - the logic should be in the parent function but for historical reasons it is randomly spread across report classes)